lib: lte_lc: add DNS fallback (#21352)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

* :ref:`lte_lc_readme` library:

  * Added:

>     * The sending of ``LTE_LC_EVT_NEIGHBOR_CELL_MEAS`` event with ``current_cell`` set to ``LTE_LC_CELL_EUTRAN_ID_INVALID`` in case an error occurs while parsing the ``%NCELLMEAS`` notification.

Reviewer: ```suggestion
    * Sending of the ``LTE_LC_EVT_NEIGHBOR_CELL_MEAS`` event with ``current_cell`` set to ``LTE_LC_CELL_EUTRAN_ID_INVALID`` if an error occurs while parsing the ``%NCELLMEAS`` notification.
```

Author: Updated

>     * The :kconfig:option:`CONFIG_LTE_LC_DNS_FALLBACK_MODULE` and :kconfig:option:`CONFIG_LTE_LC_DNS_FALLBACK_ADDRESS` options to enable setting a fallback DNS address.

Reviewer: ```suggestion
    * The :kconfig:option:`CONFIG_LTE_LC_DNS_FALLBACK_MODULE` and :kconfig:option:`CONFIG_LTE_LC_DNS_FALLBACK_ADDRESS` Kconfig options to enable setting a fallback DNS address.
```

Author: Thanks

>       This option is enabled by default. If the application has configured a DNS server address in Zephyr's native networking stack, using the :kconfig:option:`CONFIG_DNS_SERVER1` Kconfig option,

Reviewer 2: ```suggestion
      The :kconfig:option:`CONFIG_LTE_LC_DNS_FALLBACK_MODULE` Kconfig option is enabled by default. If the application has configured a DNS server address in Zephyr's native networking stack, using the :kconfig:option:`CONFIG_DNS_SERVER1` Kconfig option,
```

>       This option is enabled by default. If the application has configured a DNS server address in Zephyr's native networking stack, using the :kconfig:option:`CONFIG_DNS_SERVER1` Kconfig option,
>       the same server is set as the fallback address for DNS queries offloaded to the nRF9 series modem. Otherwise, the :kconfig:option:`CONFIG_LTE_LC_DNS_FALLBACK_ADDRESS` Kconfig option
>       controls the fallback DNS server address, which by default is set to Cloudflare's DNS server: 1.1.1.1.
>       The device may or may not receive a DNS address by the network during PDN connection.
>       Even within the same network, the PDN connection establishment method (PCO vs ePCO) may change when the device operates in NB-IoT or LTE Cat-M1, resulting in missing DNS addresses when one method is used, but not the other.

Reviewer: ```suggestion
      This option is enabled by default.
      If the application has configured a DNS server address in Zephyr's native networking stack, using the :kconfig:option:`CONFIG_DNS_SERVER1` Kconfig option, the same server is set as the fallback address for DNS queries offloaded to the nRF9 Series modem.
      Otherwise, the :kconfig:option:`CONFIG_LTE_LC_DNS_FALLBACK_ADDRESS` Kconfig option controls the fallback DNS server address that is set to Cloudflare's DNS server: 1.1.1.1 by default.
      The device might or might not receive a DNS address by the network during PDN connection.
      Even within the same network, the PDN connection establishment method (PCO vs ePCO) might change when the device operates in NB-IoT or LTE Cat-M1, resulting in missing DNS addresses when one method is used, but not the other.
```
You have two Kconfig options, but say "this" option. Which one are you referring to?
And the same question about the nRF9 Series as above.

Author: Added specific Kconfig option I was referring to, and updated text.
